gradle/projectDependencyGraph.gradle

```diff
@@ -47,13 +47,15 @@ task projectDependencyGraph {
                 .withType(ProjectDependency)
                 .collect { it.dependencyProject }
                 .each { dependency ->
-          projects.add(project)
-          projects.add(dependency)
-          rootProjects.remove(dependency)
-
-          def graphKey = new Tuple2<Project, Project>(project, dependency)
-          def traits = dependencies.computeIfAbsent(graphKey) { new ArrayList<String>() }
+          if (!(config.name.toLowerCase().contains('androidtest') || config.name.toLowerCase().contains('unittest'))) {
```

Comment:
I would say this is doing a bit more than fixing the issue:

1. The issue only happens to Android modules, so I think we could choose to do this only when we're dealing with an Android module as well.
2. I wonder if instead of skipping it entirely, we could skip only when we've found the Android test module depending on itself. I'll play around with this idea here and come back if I can find a good way to do this.

Reply:
This would do for point 1 (and would also resolve [my other comment](https://github.com/JakeWharton/SdkSearch/pull/188/files#r513269518)):

```groovy
project.configurations.findAll { config ->
  !(isAndroid(project) && isTest(config))
}.each { config ->
  // ...
}

// ...

static def isAndroid(Project project) {
  project.plugins.hasPlugin('com.android.library') || project.plugins.hasPlugin('com.android.application')
}

static def isTest(Configuration config) {
  config.name.toLowerCase().contains('androidtest') || config.name.toLowerCase().contains('unittest')
}
```

Reply:
Hi, I just upgraded my AGP and now face this self-dependency of each of my Android modules to themselves...
any chances of seeing @tfcporciuncula PR accepted?

Reply:
Hello ! Any chance getting a fix for this ?